spec/signalfx-transform-helper-spec.js

```diff
@@ -57,25 +55,27 @@ describe('signalfx-transform-helper', () => {
     };
     const flattened = signalfxHelper.toKeyValueMap(obj);
     expect(flattened['resources_0']).to.be.equal('a');
-    expect(flattened['resources_1_b']).to.be.equal('B');
-    expect(flattened['resources_2_c_c_c']).to.be.equal('C');
+    expect(flattened['resources_1']).to.be.equal('{"b":"B"}');
+    expect(flattened['resources_2']).to.be.equal('{"c":{"c":{"c":"C"}}}');
     expect(flattened['resources_3']).to.be.equal('d');
-    expect(flattened['resources_4_e_0']).to.be.equal(1);
-    expect(flattened['resources_4_e_1']).to.be.equal(2);
-    expect(flattened['resources_4_e_2']).to.be.equal(3);
+    expect(flattened['resources_4']).to.be.equal('{"e":[1,2,3]}');
   });
 
   it('should handle sample Amazon event', () => {
     const event = require('./amazonAlert.json');
     const flattened = signalfxHelper.toKeyValueMap(event);
     expect(flattened['detail_risk-score']).to.be.equal(80);
-    expect(flattened['detail_summary_IP_34_199_185_34']).to.be.equal(121);
-    expect(flattened['detail_summary_Time_Range_0_end']).to.be.equal('2017-04-24T20:25:54Z');
+    let expectedDetailSummary = JSON.stringify(require('./expectedDetailSummary.json'));
+    expect(flattened['detail_summary']).to.be.equal(expectedDetailSummary);
   });
 
-  it('should not include empty fields', () => {
+  it('should not include empty first-level fields', () => {
```

Comment:
I would probably try to unify it, i.e. either always include empty fields or never do that. I do not have strong preference which is better, I am not familiar with any use-case where this would matter, but current behavior is probably a bit tricky. So perhaps go with whichever is easier to implement, unless other reviewers have some thoughts?